src/box/xlog.c

```diff
@@ -696,6 +696,23 @@ xdir_collect_garbage(struct xdir *dir, int64_t signature, unsigned flags)
 	}
 }
 
+int
+xdir_remove_file_by_vclock(struct xdir *dir, struct vclock *to_remove)
+{
+	to_remove = vclockset_psearch(&dir->index, to_remove);
+	if (to_remove == NULL)
+		return -1;
+	const char *filename =
+		xdir_format_filename(dir, vclock_sum(to_remove), NONE);
+	int rc = unlink(filename);
+	xdir_say_gc(unlink(filename), errno, filename);
```

Comment:
1. This is a bug. I should have passed `rc` to xdir_say_gc() instead of invoking unlink one more time. Fixed.
2. Indeed seems to be a bug. I've fixed it in a separate patch and included it to PR (I guess it's ok since I anyway refactoring this part of xlog codebase).